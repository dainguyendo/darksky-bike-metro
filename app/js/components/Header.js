import React, { Component } from 'react';
import * as Colors from 'material-ui/styles/colors';

import Train from 'material-ui/svg-icons/maps/train';
import Bike from 'material-ui/svg-icons/maps/directions-bike';

const titleContainerStyle = {
  color: Colors.red500
};

const iconStyle = {
  width: '100px',
  height: '100px'
};

export default class Header extends Component {
  render () {
    return (
      <div id='header'>
        <div className='title' style={ titleContainerStyle }>
          <Bike style={ iconStyle } />
          <span>||</span>
          <Train style={ iconStyle } />
        </div>
      </div>
    );
  }
}
