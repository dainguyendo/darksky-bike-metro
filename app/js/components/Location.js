import React, { Component } from 'react';

import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';

import store from 'js/redux/store';
import { saveLatitude, saveLongitude } from 'js/redux/actions/criteriaActions';


export default class Location extends Component {
  constructor () {
    super();
    this.state = {
      input: ''
    };
  }

  render () {
    return (
      <div id='location'>
        <TextField
          hintText='Enter your destination'
          defaultValue='6856 Alicia Court, Alexandria, VA 22310'
          onChange={ this.handleInputChange }/>
        <RaisedButton
          label='Search'
          onClick={ this.searchAddress } />
      </div>
    );
  }

  handleInputChange = (event, newValue) => {
    this.setState({ input: newValue });
  }

  searchAddress = () => {
    const base = 'https://maps.googleapis.com/maps/api/geocode/json?address=';
    const address = this.state.input.split(' ').join('+');
    const key = `&key=${process.env.GOOGLE_GEOCODE_API_KEY}`;

    const url = base + address + key;

    const xmlhttp = new XMLHttpRequest();
    xmlhttp.onreadystatechange = () => {
      // Request is finished
      if (xmlhttp.readyState === 4) {
        // Request is OK
        if (xmlhttp.status === 200) {
          // Grab first return
          const result = xmlhttp.response.results[0];
          const { lat, lng } = result.geometry.location;
          const formattedAddress = result.formatted_address;

          console.log(lat, lng, formattedAddress);
          store.dispatch(saveLatitude(lat));
          store.dispatch(saveLongitude(lng));
        }
      }
    };

    xmlhttp.open('GET', url);
    xmlhttp.responseType = 'json';
    xmlhttp.send();
  }
}
