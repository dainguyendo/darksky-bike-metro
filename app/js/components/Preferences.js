import React, { Component } from 'react';

import Temperature from 'js/components/preferences/Temperature';
import Rain from 'js/components/preferences/Rain';
import Date from 'js/components/preferences/Date';

import RaisedButton from 'material-ui/RaisedButton';

export default class Preferences extends Component {
  constructor () {
    super();
    this.state = { displayPreferences: false };
  }

  render () {
    const show = this.state.displayPreferences ? '' : 'hidden';
    return (
      <div id='preference' className='column-container' >
        <RaisedButton
          className='btn'
          label='Configure ride conditions'
          onClick={ this.togglePreferences } />
        <div className={`column-container ${show}`}>
          <Temperature minTemperature={ this.props.minTemperature } maxTemperature={ this.props.maxTemperature } />
          <Rain chanceOfRain={ this.props.chanceOfRain } />
          <Date />
        </div>
      </div>
    );
  }

  togglePreferences = () => { this.setState({ displayPreferences: !this.state.displayPreferences }); }
}
