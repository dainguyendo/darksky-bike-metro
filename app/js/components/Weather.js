import React, { Component } from 'react';
import * as Colors from 'material-ui/styles/colors';
import {
  Table,
  TableBody,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';

import { isEmpty } from 'js/utils/utilities';

const tableBodyStyle = {
  overflow: 'scroll',
};

const fontBolderColorStyle = {
  fontWeight: 'bolder',
  color: Colors.grey50,
  fontSize: '14px',
  paddingLeft: '5px',
  paddingRight: '5px'
};

const fontColorStyle = {
  color: Colors.grey50,
  fontSize: '14px',
  paddingLeft: '5px',
  paddingRight: '5px'
};

const tableStyle = {
  backgroundColor: 'rgba(0,0,0,0)'
};

export default class Weather extends Component {
  render () {
    const { result } = this.props;
    let row = null;

    if (!isEmpty(result)) {
      row = Object.entries(result.summary).map((entry, idx) => {
        return (
          <TableRow key={ idx }>
            <TableRowColumn style={ fontBolderColorStyle }>{ `${entry[0]}` }</TableRowColumn>
            <TableRowColumn style={ fontColorStyle }>{ entry[1] }</TableRowColumn>
          </TableRow>
        );
      });
    }

    return (
      <Table
        fixedHeader={false}
        fixedFooter={false}
        height='100%'
        selectable={ false }
        style={ tableStyle }>
        <TableBody style={ tableBodyStyle } displayRowCheckbox={ false }>
          { row }
        </TableBody>
      </Table>
    );
  }
}
