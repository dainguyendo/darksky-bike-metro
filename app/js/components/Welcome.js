import React, { Component } from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import Help from 'material-ui/svg-icons/action/help';

export default class Welcome extends Component {
  constructor () {
    super();
    this.state = { open: false };
  }
  render () {
    const actions = [
      <FlatButton
        label='Close'
        primary={ true }
        onClick={this.handleClose} />,
    ];

    return (
      <div>
        <FlatButton
          icon={ <Help /> }
          className='btn'
          label='Getting started'
          onClick={ this.handleOpen } />
        <Dialog
          className='subtitle'
          title='Welcome'
          actions={ actions }
          modal={false}
          open={ this.state.open }
          onRequestClose={ this.handleClose } >
          <div className='paragraph'>
            Commuting choices are tough sometimes, especially when weather comes into play. Use this application to help make your choice easier! Simply search your commute address and see if you should bike or take the train. A little picky? Use 'Configure Ride Conditions' to set the specifics!
          </div>
        </Dialog>
      </div>
    );
  }

  handleOpen = () => { this.setState({ open: true }); };

  handleClose = () => { this.setState({ open: false }); };
}
