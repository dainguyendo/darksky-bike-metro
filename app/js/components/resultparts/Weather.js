import React, { Component } from 'react';
import * as Colors from 'material-ui/styles/colors';
import {
  Table,
  TableBody,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';

import { isEmpty } from 'js/utils/utilities';

const tableBodyStyle = {
  overflow: 'scroll',
};

const fontBolderColorStyle = {
  fontWeight: 'bolder',
  color: Colors.grey50
};

const fontColorStyle = {
  color: Colors.grey50
};

const tableStyle = {
  backgroundColor: 'rgba(0,0,0,0)'
};

export default class Weather extends Component {
  render () {
    const { result } = this.props;
    let row = null;

    if (!isEmpty(result)) {
      row = Object.entries(result.summary).map(entry => {
        return (
          <TableRow>
            <TableRowColumn style={ fontBolderColorStyle }>{ entry[0] }</TableRowColumn>
            <TableRowColumn style={ fontColorStyle }>{ entry[1] }</TableRowColumn>
          </TableRow>
        );
      });
    }

    return (
      <Table
        fixedHeader={false}
        fixedFooter={false}
        height='100%'
        selectable={ false }
        style={ tableStyle }>
        <TableBody style={ tableBodyStyle } displayRowCheckbox={ false }>
          { row }
        </TableBody>
      </Table>
    );
  }
}
