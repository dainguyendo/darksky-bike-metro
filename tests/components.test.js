import React from 'react';
import { shallow } from 'enzyme';

const testState = {
  'chanceOfRain': 50,
  'latitude': 38.8935755,
  'longitude': -77.0846155,
  'minTemperature': 32,
  'maxTemperature': 100,
  'data': {},
  'unixTime': 1506250800,
  'result': {},
  'drawerVisible': false
};

// React Components
import Header from '../app/js/components/Header';
import Location from '../app/js/components/Location';
import Preferences from '../app/js/components/Preferences';
import Weather from '../app/js/components/Weather';
import Welcome from '../app/js/components/Welcome';

test('Header matches snapshot', () => {
  const component = shallow(<Header />);
  expect(component).toMatchSnapshot();
});

test('Location matches snapshot', () => {
  const component = shallow(<Location />);
  expect(component).toMatchSnapshot();
});

test('Preferences matches snapshot', () => {
  const component = shallow(<Preferences {...testState} />);
  expect(component).toMatchSnapshot();
});

test('Weather matches snapshot', () => {
  const component = shallow(<Weather result={testState.result} />);
  expect(component).toMatchSnapshot();
});

test('Welcome matches snapshot', () => {
  const component = shallow(<Welcome />);
  expect(component).toMatchSnapshot();
});
