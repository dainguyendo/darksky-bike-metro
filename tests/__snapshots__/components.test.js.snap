// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    id="header"
>
    <div
        className="title"
        style={
            Object {
                "color": "#f44336",
              }
        }
    >
        <MapsDirectionsBike
            style={
                Object {
                    "height": "100px",
                    "width": "100px",
                  }
            }
        />
        <span>
            ||
        </span>
        <MapsTrain
            style={
                Object {
                    "height": "100px",
                    "width": "100px",
                  }
            }
        />
    </div>
</div>,
  "nodes": Array [
    <div
      id="header"
>
      <div
            className="title"
            style={
                  Object {
                        "color": "#f44336",
                      }
            }
      >
            <MapsDirectionsBike
                  style={
                        Object {
                              "height": "100px",
                              "width": "100px",
                            }
                  }
            />
            <span>
                  ||
            </span>
            <MapsTrain
                  style={
                        Object {
                              "height": "100px",
                              "width": "100px",
                            }
                  }
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          id="header"
>
          <div
                    className="title"
                    style={
                              Object {
                                        "color": "#f44336",
                                      }
                    }
          >
                    <MapsDirectionsBike
                              style={
                                        Object {
                                                  "height": "100px",
                                                  "width": "100px",
                                                }
                              }
                    />
                    <span>
                              ||
                    </span>
                    <MapsTrain
                              style={
                                        Object {
                                                  "height": "100px",
                                                  "width": "100px",
                                                }
                              }
                    />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          id="header"
>
          <div
                    className="title"
                    style={
                              Object {
                                        "color": "#f44336",
                                      }
                    }
          >
                    <MapsDirectionsBike
                              style={
                                        Object {
                                                  "height": "100px",
                                                  "width": "100px",
                                                }
                              }
                    />
                    <span>
                              ||
                    </span>
                    <MapsTrain
                              style={
                                        Object {
                                                  "height": "100px",
                                                  "width": "100px",
                                                }
                              }
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header />,
}
`;

exports[`Location matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="column-container"
    id="location"
>
    <TextField
        className="text-field"
        defaultValue="6856 Alicia Court, Alexandria, VA 22310"
        disabled={false}
        errorText=""
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="Enter your destination"
        multiLine={true}
        onChange={[Function]}
        rows={1}
        rowsMax={3}
        type="text"
        underlineShow={true}
    />
    <RaisedButton
        className="btn"
        disabled={false}
        fullWidth={false}
        label="Search Address"
        labelPosition="after"
        onClick={[Function]}
        primary={false}
        secondary={false}
    />
</div>,
  "nodes": Array [
    <div
      className="column-container"
      id="location"
>
      <TextField
            className="text-field"
            defaultValue="6856 Alicia Court, Alexandria, VA 22310"
            disabled={false}
            errorText=""
            floatingLabelFixed={false}
            fullWidth={false}
            hintText="Enter your destination"
            multiLine={true}
            onChange={[Function]}
            rows={1}
            rowsMax={3}
            type="text"
            underlineShow={true}
      />
      <RaisedButton
            className="btn"
            disabled={false}
            fullWidth={false}
            label="Search Address"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Location />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Location {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleInputChange": [Function],
        "props": Object {},
        "refs": Object {},
        "searchAddress": [Function],
        "state": Object {
          "errorAddress": "",
          "input": "6856 Alicia Court, Alexandria, VA 22310",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="column-container"
          id="location"
>
          <TextField
                    className="text-field"
                    defaultValue="6856 Alicia Court, Alexandria, VA 22310"
                    disabled={false}
                    errorText=""
                    floatingLabelFixed={false}
                    fullWidth={false}
                    hintText="Enter your destination"
                    multiLine={true}
                    onChange={[Function]}
                    rows={1}
                    rowsMax={3}
                    type="text"
                    underlineShow={true}
          />
          <RaisedButton
                    className="btn"
                    disabled={false}
                    fullWidth={false}
                    label="Search Address"
                    labelPosition="after"
                    onClick={[Function]}
                    primary={false}
                    secondary={false}
          />
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="column-container"
          id="location"
>
          <TextField
                    className="text-field"
                    defaultValue="6856 Alicia Court, Alexandria, VA 22310"
                    disabled={false}
                    errorText=""
                    floatingLabelFixed={false}
                    fullWidth={false}
                    hintText="Enter your destination"
                    multiLine={true}
                    onChange={[Function]}
                    rows={1}
                    rowsMax={3}
                    type="text"
                    underlineShow={true}
          />
          <RaisedButton
                    className="btn"
                    disabled={false}
                    fullWidth={false}
                    label="Search Address"
                    labelPosition="after"
                    onClick={[Function]}
                    primary={false}
                    secondary={false}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Location />,
}
`;

exports[`Preferences matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="column-container"
    id="preference"
>
    <RaisedButton
        className="btn"
        disabled={false}
        fullWidth={false}
        label="Configure ride conditions"
        labelPosition="after"
        onClick={[Function]}
        primary={false}
        secondary={false}
    />
    <div
        className="column-container hidden"
    >
        <Temperature
            maxTemperature={100}
            minTemperature={32}
        />
        <Rain
            chanceOfRain={50}
        />
        <Preferences />
    </div>
</div>,
  "nodes": Array [
    <div
      className="column-container"
      id="preference"
>
      <RaisedButton
            className="btn"
            disabled={false}
            fullWidth={false}
            label="Configure ride conditions"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
      />
      <div
            className="column-container hidden"
      >
            <Temperature
                  maxTemperature={100}
                  minTemperature={32}
            />
            <Rain
                  chanceOfRain={50}
            />
            <Preferences />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Preferences
        chanceOfRain={50}
        data={Object {}}
        drawerVisible={false}
        latitude={38.8935755}
        longitude={-77.0846155}
        maxTemperature={100}
        minTemperature={32}
        result={Object {}}
        unixTime={1506250800}
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Preferences {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "chanceOfRain": 50,
          "data": Object {},
          "drawerVisible": false,
          "latitude": 38.8935755,
          "longitude": -77.0846155,
          "maxTemperature": 100,
          "minTemperature": 32,
          "result": Object {},
          "unixTime": 1506250800,
        },
        "refs": Object {},
        "state": Object {
          "displayPreferences": false,
        },
        "togglePreferences": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="column-container"
          id="preference"
>
          <RaisedButton
                    className="btn"
                    disabled={false}
                    fullWidth={false}
                    label="Configure ride conditions"
                    labelPosition="after"
                    onClick={[Function]}
                    primary={false}
                    secondary={false}
          />
          <div
                    className="column-container hidden"
          >
                    <Temperature
                              maxTemperature={100}
                              minTemperature={32}
                    />
                    <Rain
                              chanceOfRain={50}
                    />
                    <Preferences />
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="column-container"
          id="preference"
>
          <RaisedButton
                    className="btn"
                    disabled={false}
                    fullWidth={false}
                    label="Configure ride conditions"
                    labelPosition="after"
                    onClick={[Function]}
                    primary={false}
                    secondary={false}
          />
          <div
                    className="column-container hidden"
          >
                    <Temperature
                              maxTemperature={100}
                              minTemperature={32}
                    />
                    <Rain
                              chanceOfRain={50}
                    />
                    <Preferences />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Preferences
    chanceOfRain={50}
    data={Object {}}
    drawerVisible={false}
    latitude={38.8935755}
    longitude={-77.0846155}
    maxTemperature={100}
    minTemperature={32}
    result={Object {}}
    unixTime={1506250800}
/>,
}
`;

exports[`Weather matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Table
    allRowsSelected={false}
    fixedFooter={false}
    fixedHeader={false}
    height="100%"
    multiSelectable={false}
    selectable={false}
    style={
        Object {
            "backgroundColor": "rgba(0,0,0,0)",
          }
    }
>
    <TableBody
        allRowsSelected={false}
        deselectOnClickaway={true}
        displayRowCheckbox={false}
        multiSelectable={false}
        preScanRows={true}
        selectable={true}
        style={
            Object {
                "overflow": "scroll",
              }
        }
    />
</Table>,
  "nodes": Array [
    <Table
      allRowsSelected={false}
      fixedFooter={false}
      fixedHeader={false}
      height="100%"
      multiSelectable={false}
      selectable={false}
      style={
            Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                }
      }
>
      <TableBody
            allRowsSelected={false}
            deselectOnClickaway={true}
            displayRowCheckbox={false}
            multiSelectable={false}
            preScanRows={true}
            selectable={true}
            style={
                  Object {
                        "overflow": "scroll",
                      }
            }
      />
</Table>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Weather
        result={Object {}}
/>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Weather {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "result": Object {},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Table
          allRowsSelected={false}
          fixedFooter={false}
          fixedHeader={false}
          height="100%"
          multiSelectable={false}
          selectable={false}
          style={
                    Object {
                              "backgroundColor": "rgba(0,0,0,0)",
                            }
          }
>
          <TableBody
                    allRowsSelected={false}
                    deselectOnClickaway={true}
                    displayRowCheckbox={false}
                    multiSelectable={false}
                    preScanRows={true}
                    selectable={true}
                    style={
                              Object {
                                        "overflow": "scroll",
                                      }
                    }
          />
</Table>,
        "_debugID": 8,
        "_renderedOutput": <Table
          allRowsSelected={false}
          fixedFooter={false}
          fixedHeader={false}
          height="100%"
          multiSelectable={false}
          selectable={false}
          style={
                    Object {
                              "backgroundColor": "rgba(0,0,0,0)",
                            }
          }
>
          <TableBody
                    allRowsSelected={false}
                    deselectOnClickaway={true}
                    displayRowCheckbox={false}
                    multiSelectable={false}
                    preScanRows={true}
                    selectable={true}
                    style={
                              Object {
                                        "overflow": "scroll",
                                      }
                    }
          />
</Table>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Weather
    result={Object {}}
/>,
}
`;

exports[`Welcome matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <FlatButton
        className="btn"
        disabled={false}
        fullWidth={false}
        icon={<ActionHelp />}
        label="Getting started"
        labelPosition="after"
        labelStyle={Object {}}
        onClick={[Function]}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={false}
    />
    <Dialog
        actions={
            Array [
                <FlatButton
                  disabled={false}
                  fullWidth={false}
                  label="Close"
                  labelPosition="after"
                  labelStyle={Object {}}
                  onClick={[Function]}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  primary={true}
                  secondary={false}
            />,
              ]
        }
        autoDetectWindowHeight={true}
        autoScrollBodyContent={false}
        className="subtitle"
        modal={false}
        onRequestClose={[Function]}
        open={false}
        repositionOnUpdate={true}
        title="Welcome"
    >
        <div
            className="paragraph"
        >
            Commuting choices are tough sometimes, especially when weather comes into play. Use this application to help make your choice easier! Simply search your commute address and see if you should bike or take the train. A little picky? Use 'Configure Ride Conditions' to set the specifics!
        </div>
    </Dialog>
</div>,
  "nodes": Array [
    <div>
      <FlatButton
            className="btn"
            disabled={false}
            fullWidth={false}
            icon={<ActionHelp />}
            label="Getting started"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
      />
      <Dialog
            actions={
                  Array [
                        <FlatButton
                          disabled={false}
                          fullWidth={false}
                          label="Close"
                          labelPosition="after"
                          labelStyle={Object {}}
                          onClick={[Function]}
                          onKeyboardFocus={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onTouchStart={[Function]}
                          primary={true}
                          secondary={false}
                  />,
                      ]
            }
            autoDetectWindowHeight={true}
            autoScrollBodyContent={false}
            className="subtitle"
            modal={false}
            onRequestClose={[Function]}
            open={false}
            repositionOnUpdate={true}
            title="Welcome"
      >
            <div
                  className="paragraph"
            >
                  Commuting choices are tough sometimes, especially when weather comes into play. Use this application to help make your choice easier! Simply search your commute address and see if you should bike or take the train. A little picky? Use 'Configure Ride Conditions' to set the specifics!
            </div>
      </Dialog>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Welcome />,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Welcome {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleClose": [Function],
        "handleOpen": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "open": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <FlatButton
                    className="btn"
                    disabled={false}
                    fullWidth={false}
                    icon={<ActionHelp />}
                    label="Getting started"
                    labelPosition="after"
                    labelStyle={Object {}}
                    onClick={[Function]}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onTouchStart={[Function]}
                    primary={false}
                    secondary={false}
          />
          <Dialog
                    actions={
                              Array [
                                        <FlatButton
                                          disabled={false}
                                          fullWidth={false}
                                          label="Close"
                                          labelPosition="after"
                                          labelStyle={Object {}}
                                          onClick={[Function]}
                                          onKeyboardFocus={[Function]}
                                          onMouseEnter={[Function]}
                                          onMouseLeave={[Function]}
                                          onTouchStart={[Function]}
                                          primary={true}
                                          secondary={false}
                              />,
                                      ]
                    }
                    autoDetectWindowHeight={true}
                    autoScrollBodyContent={false}
                    className="subtitle"
                    modal={false}
                    onRequestClose={[Function]}
                    open={false}
                    repositionOnUpdate={true}
                    title="Welcome"
          >
                    <div
                              className="paragraph"
                    >
                              Commuting choices are tough sometimes, especially when weather comes into play. Use this application to help make your choice easier! Simply search your commute address and see if you should bike or take the train. A little picky? Use 'Configure Ride Conditions' to set the specifics!
                    </div>
          </Dialog>
</div>,
        "_debugID": 10,
        "_renderedOutput": <div>
          <FlatButton
                    className="btn"
                    disabled={false}
                    fullWidth={false}
                    icon={<ActionHelp />}
                    label="Getting started"
                    labelPosition="after"
                    labelStyle={Object {}}
                    onClick={[Function]}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onTouchStart={[Function]}
                    primary={false}
                    secondary={false}
          />
          <Dialog
                    actions={
                              Array [
                                        <FlatButton
                                          disabled={false}
                                          fullWidth={false}
                                          label="Close"
                                          labelPosition="after"
                                          labelStyle={Object {}}
                                          onClick={[Function]}
                                          onKeyboardFocus={[Function]}
                                          onMouseEnter={[Function]}
                                          onMouseLeave={[Function]}
                                          onTouchStart={[Function]}
                                          primary={true}
                                          secondary={false}
                              />,
                                      ]
                    }
                    autoDetectWindowHeight={true}
                    autoScrollBodyContent={false}
                    className="subtitle"
                    modal={false}
                    onRequestClose={[Function]}
                    open={false}
                    repositionOnUpdate={true}
                    title="Welcome"
          >
                    <div
                              className="paragraph"
                    >
                              Commuting choices are tough sometimes, especially when weather comes into play. Use this application to help make your choice easier! Simply search your commute address and see if you should bike or take the train. A little picky? Use 'Configure Ride Conditions' to set the specifics!
                    </div>
          </Dialog>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Welcome />,
}
`;
